src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetMempoolSize(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Invalid memory pool size");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mempool_size = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetMempoolCacheSize(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0 || entry_int > RTE_MEMPOOL_CACHE_MAX_SIZE) {
+        SCLogError(SC_ERR_INVALID_VALUE,
+                "Invalid memory pool cache size"
+                "(max %" PRIu32,
+                RTE_MEMPOOL_CACHE_MAX_SIZE);
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mempool_cache_size = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetRxDescriptors(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Number of RX descriptors must be a positive number");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->nb_rx_desc = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetTxDescriptors(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int <= 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Number of TX descriptors must be a positive number");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->nb_rx_desc = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetMtu(DPDKIfaceConfig *iconf, intmax_t entry_int)
+{
+    if (entry_int < RTE_ETHER_MIN_MTU || entry_int > RTE_ETHER_MAX_JUMBO_FRAME_LEN) {
+        SCLogError(SC_ERR_DPDK_CONF, "Size of MTU must be between %" PRIu32 " and %" PRIu32,
+                RTE_ETHER_MIN_MTU, RTE_ETHER_MAX_JUMBO_FRAME_LEN);
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    iconf->mtu = entry_int;
+    SCReturnInt(0);
+}
+
+int ConfigSetPromiscuousMode(DPDKIfaceConfig *iconf, int entry_bool)
+{
+    if (entry_bool)
+        iconf->flags |= DPDK_PROMISC;
+
+    SCReturnInt(0);
+}
+
+int ConfigSetMulticast(DPDKIfaceConfig *iconf, int entry_bool)
+{
+    if (entry_bool)
+        iconf->flags |= DPDK_MULTICAST; // enable
+
+    SCReturnInt(0);
+}
+
+int ConfigSetChecksumChecks(DPDKIfaceConfig *iconf, int entry_bool)
+{
+    if (entry_bool)
+        iconf->checksum_mode = CHECKSUM_VALIDATION_ENABLE;
+
+    SCReturnInt(0);
+}
+
+int ConfigSetChecksumOffload(DPDKIfaceConfig *iconf, int entry_bool)
+{
+    if (entry_bool)
+        iconf->flags |= DPDK_RX_CHECKSUM_OFFLOAD;
+
+    SCReturnInt(0);
+}
+
+int ConfigSetCopyIface(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    int retval;
+
+    if (entry_str == NULL || entry_str[0] == '\0' || strcmp(entry_str, "none") == 0) {
+        iconf->out_iface = NULL;
+        SCReturnInt(0);
+    }
+
+    retval = rte_eth_dev_get_port_by_name(entry_str, &iconf->out_port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Name of the copy interface (%s) not valid", entry_str);
```

Comment:
`errno` value might be helpful here.

Reply:
is using `errno` this way a common pattern in DPDK apps? I normally leave it alone as its set by libc functionally generally.

Reply:
My comment here (and in other similar places) is to expose the errno value (or string) in the message.

Reply:
After comments from Jeff I have reworked the error handling logic and now I am using mostly a `FatalError` approach.